K2Bridge/Visitors/Aggregations/MetricAggregationVisitor.cs

```diff
@@ -68,6 +79,8 @@ public void Visit(ExtendedStatsAggregation extendedStatsAggregation)
             query.Append(")");
 
             extendedStatsAggregation.KustoQL = query.ToString();
+
+            VisitedMetrics.Add(keyWithMetadata);
```

Comment:
shouldn't it be .Add(key)?

Reply:
as line 94

Reply:
When I 'propagate' the metrics from on query to the other, I need also propagate the metadata. I dislike this code, and I suggest we extend the work from @tomconte on Ranges to pass all metadata in a separate table and no longer with keys. It will simplify this code and avoid to call VisitedMetrics on each individual visitor.